src/containers/Dashboard/index.js

```diff
@@ -0,0 +1,233 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
+import * as actions from '../../state/dashboard/actions';
+import * as chartSelectors from '../../state/dashboard/reducer';
+import DashboardSection from './DashboardSection';
+import TimeRangeSelect from '../../components/TimeRangeSelect';
+
+import './Dashboard.scss';
+
+class Dashboard extends Component {
+  componentWillMount() {
+    const {
+      timeOptions: { range },
+      updatedTimeRange,
+      fetchDashboardData
+    } = this.props;
+
+    fetchDashboardData();
+
+    // data live reloads every 5 seconds
+    this.intervalId = setInterval(
+      (function fetchData() {
+        updatedTimeRange(range);
+        return fetchData;
+      })(),
```

Comment:
This idea of invoking the callback function immediately is interesting. I never thought it could be done like that. Is it a common pattern in JS?

Reply:
Hmm, I just know I've seen it used before for this purpose, so I've begun using it myself :)